# first i am setting the folder i will be working in 

setwd("/Users/iamdeanooo/Documents/GitHub/Bioinformatics/")
# then i am reading the fasta file in the folder and giving it a variable
sequences <- readDNAStringSet("sequences.fasta")

# this is singling out whichever sequence I want
sequences$Homo_sapiens_20

myAAseqs <- AAStringSet(sequences)
myAlignment <- msa(myAAseqs, "ClustalW")
myAlignment

# after running my alignment i can see homo_sapiens_6 has two mismatches and a deletion
# before the ... which is AT when most of the other sequences end in CA

if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install()
BiocManager::install("Biostrings")
BiocManager::install("DECIPHER")
library(Biostrings)
library(DECIPHER)
alignment <- pairwiseAlignment("sequences.fasta")
msaIdentity("sequences.fasta")
pairwiseAlignment("sequences.fasta")
library(Biostrings)
sequences <- pairwiseAlignment(sequences)
print(myAlignment)
print(identi)
writeXStringSet("AATCTACTCCCAGGAGCAGGGAGGGCAGGAGCCAGG
                GCTGGGCATGAAAGTCAGGGCAGAGCCATCTATTGCTTACATTTGCTTCTGACACAACTGTGTTCACTAG
                CAACCTCAAACAGACACCATGGTGCACCTGACTCCTGTGGAGAAGTCTGCCGTTACTGCCCTGTGGGGCA
                AGGTGAACGTGGATGAAGTTGGTGGTGAGGCCCTGGGCAGGTTGGTATCAAGGTTACAAGACAGGTTTAA
                GGAGACCAATAGAAACTGGGCATGTGGAGACAGAGAAGACTCTTGGGTTTCTGATAGGCACTGACTCTCT
                CTGCCTATTGGTCTATTTTCCCACCCTTAGGCTGCTGGTGGTCTACCCTTGGACCCAGAGGTTCTTTGAG
                TCCTTTGGGGATCTGTCCACTCCTGATGCTGTTATGGGCAACCCTAAGGTGAAGGCTCATGGCAAGAAAG
                TGCTCGGTGCCTTTAGTGATGGCCTGGCTCACCTGGACAACCTCAAGGGCACCTTTGCCACACTGAGTGA
                GCTGCACTGTGACAAGCTGCACGTGGATCCTGAGAACTTCAGGGTGAGTCTATGGGACCCTTGATGTTTT
                CTTTCCCCTTCTTTTCTATGGTTAAGTTCATGTCATAGGAAGGGG)
# i am now converting my alignment to seqinr                
                
dis_align <- msaConvert(myAlignment, type = "seqinr::alignment")

# now i am running a distance alignment
dismat <- dist.alignment(dis_align)

dna_sequences <- sequences
protein.sequence <- translate(DNAString(dna_sequence)
print(protein.sequence)
dna_sequence <- (AATCTACTCCCAGGAGCAGGGAGGGCAGGAGCCAGG
                GCTGGGCATGAAAGTCAGGGCAGAGCCATCTATTGCTTACATTTGCTTCTGACACAACTGTGTTCACTAG
                CAACCTCAAACAGACACCATGGTGCACCTGACTCCTGTGGAGAAGTCTGCCGTTACTGCCCTGTGGGGCA
                AGGTGAACGTGGATGAAGTTGGTGGTGAGGCCCTGGGCAGGTTGGTATCAAGGTTACAAGACAGGTTTAA
                GGAGACCAATAGAAACTGGGCATGTGGAGACAGAGAAGACTCTTGGGTTTCTGATAGGCACTGACTCTCT
                CTGCCTATTGGTCTATTTTCCCACCCTTAGGCTGCTGGTGGTCTACCCTTGGACCCAGAGGTTCTTTGAG
                TCCTTTGGGGATCTGTCCACTCCTGATGCTGTTATGGGCAACCCTAAGGTGAAGGCTCATGGCAAGAAAG
                TGCTCGGTGCCTTTAGTGATGGCCTGGCTCACCTGGACAACCTCAAGGGCACCTTTGCCACACTGAGTGA
                GCTGCACTGTGACAAGCTGCACGTGGATCCTGAGAACTTCAGGGTGAGTCTATGGGACCCTTGATGTTTT
                CTTTCCCCTTCTTTTCTATGGTTAAGTTCATGTCATAGGAAGGGG)

> AATCTACTCCCAGGAGCAGGGAGGGCAGGAGCCAGG
                GCTGGGCATGAAAGTCAGGGCAGAGCCATCTATTGCTTACATTTGCTTCTGACACAACTGTGTTCACTAG
                CAACCTCAAACAGACACCATGGTGCACCTGACTCCTGTGGAGAAGTCTGCCGTTACTGCCCTGTGGGGCA
                AGGTGAACGTGGATGAAGTTGGTGGTGAGGCCCTGGGCAGGTTGGTATCAAGGTTACAAGACAGGTTTAA
                GGAGACCAATAGAAACTGGGCATGTGGAGACAGAGAAGACTCTTGGGTTTCTGATAGGCACTGACTCTCT
                CTGCCTATTGGTCTATTTTCCCACCCTTAGGCTGCTGGTGGTCTACCCTTGGACCCAGAGGTTCTTTGAG
                TCCTTTGGGGATCTGTCCACTCCTGATGCTGTTATGGGCAACCCTAAGGTGAAGGCTCATGGCAAGAAAG
                TGCTCGGTGCCTTTAGTGATGGCCTGGCTCACCTGGACAACCTCAAGGGCACCTTTGCCACACTGAGTGA
                GCTGCACTGTGACAAGCTGCACGTGGATCCTGAGAACTTCAGGGTGAGTCTATGGGACCCTTGATGTTTT
                CTTTCCCCTTCTTTTCTATGGTTAAGTTCATGTCATAGGAAGGGG
                
                